print_designer/print_designer/page/print_designer/jinja/header_footer.html

```diff
@@ -3,6 +3,7 @@
 	<head>
 		<meta charset="utf-8">
 		<link rel="preconnect" href="https://fonts.gstatic.com" />
+		<script src="https://cdn.jsdelivr.net/npm/jsbarcode@3.11.5/dist/JsBarcode.all.min.js"></script>
```

Comment:
I haven't tested this yet.

Not sure how to add the JsBarcode from /assets/... I'm always getting UnknownProtocalError during PDF generation.

Reply:
iirc js is disabled in Wkhtmltopdf

Reply:
hmmm, maybe I'll look into some python library to generate barcode SVG server side.

Reply:
Actually, it seems JsBarcode can be used in a node environment.

https://github.com/lindell/JsBarcode
```js
const { DOMImplementation, XMLSerializer } = require('xmldom');
const xmlSerializer = new XMLSerializer();
const document = new DOMImplementation().createDocument('http://www.w3.org/1999/xhtml', 'html', null);
const svgNode = document.createElementNS('http://www.w3.org/2000/svg', 'svg');

JsBarcode(svgNode, 'test', {
    xmlDocument: document,
});

const svgText = xmlSerializer.serializeToString(svgNode);
```

Maybe I can do something like this to inject the svgText into the jinja template.

Reply:
It would be better to use same library to increase the chances of having same results in pdf as seen in print designer.

Reply:
> WIP
> 
> * [x]  Spawn JsBarcode SVG in Designer
> * [x]  Modify position of SVG.
> * [ ]  Modify size of SVG.
> * [x]  Change dynamicText content
> * [ ]  JsBarcode properties in property pane
> * [x]  pdf generation
>   **Important**: Currently PDF generation only works if `options['enable-javascript'] = True` is set here:
>   https://github.com/frappe/frappe/blob/a254a256f872abdaa81183f237aca978403bd8d3/frappe/utils/pdf.py#L62-L63

what is the correct syntax of `options['enable-javascript'] = True 
for 
filedata = pdfkit.from_string(html, options=options or {}, verbose=True)

filedata = pdfkit.from_string(html, options['enable-javascript']=True or {}, verbose=True) ?

Reply:
@lstbrian before the call
`filedata = pdfkit.from_string(html, options=options or {}, verbose=True)`

add a new line above
`options['enable-javascript'] = True`